### PR TITLE
fix: fix full height issue of modal on mobile, remove useless h-modal…

### PR DESCRIPTION
**Pull request by letoast**
… class from modal

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] I have checked the page with https://validator.unl.edu/
- [ ] All the tests have passed
- [x] My pull request is based on the latest commit (not the npm version).

## ℹ Additional Information

Fixed height issues on mobile, also removed the useless h-modal class that adds a strange height: calc(100% - 2rem), which is being overwritten anyway on md+ devices...
before fix (notice the not full height on the bottom):
image

before fix (the modal on mobile is 100% height, even though it doesn't need to be):
image

### CODE REVIEW

**Comment by jjagielka** [comment]
This is not a bug. See the #457

**Comment by letoast** [comment]
I'm sorry, but it is a bug, because this library's implementation of the modal backdrop is not as per Flowbite's specification.
Yes, the 'h-modal' class is in Flowbite's class list on the modal, but the backdrop is completely differently implemented. In Flowbite's original implementation the backdrop is generated on top of the modal(in regards to the DOM), but with a lower z-index than the modal itself. In this library's implementation the backdrop is part of the modal's html element...
Proof:
Flowbite:
image
image

This library(Flowbite-svelte):
image

So either a fix is needed to sync with Flowbite's implementation or a change in the layout where this visual error would not occur any longer.

File: src/lib/modals/Modal.svelte [hunk 1]
```diff
@@ -124,7 +124,7 @@
     use:prepareFocus
     use:focusTrap
     on:click={autoclose ? onAutoClose : null}>
-    <div class="flex p-4 w-full {sizes[size]} h-full md:h-auto max-h-screen">
```
Comment (dennismphil) [inline src/lib/modals/Modal.svelte:127]: Thanks for the fix. I encountered a similar issue while using the Modal and opened an issue #523 

This PR tackles the height. I believe the same issue exists for the width as well.

Wondering if the `width` can also be fixed.

**Comment by jjagielka** [comment]
OK. I admit there was a problem in fact. Based on you PR I have introduces a few other layout tweaks in the PR #526.
@letoast can you please test it out?

**Comment by letoast** [comment]
Closed in favour of #526

**Pull request closed without merging**